docs/reference/components/sources/exec.cue

```diff
@@ -0,0 +1,231 @@
+package metadata
+
+components: sources: exec: {
+	title: "Exec"
+
+	classes: {
+		commonly_used: false
+		delivery:      "at_least_once"
+		deployment_roles: ["sidecar"]
+		development:   "beta"
+		egress_method: "stream"
+		stateful:      false
+	}
+
+	features: {
+		multiline: enabled: false
+		receive: {
+			from: {
+				service: services.exec
+			}
+
+			tls: enabled: false
+		}
+	}
+
+	support: {
+		targets: {
+			"aarch64-unknown-linux-gnu":      true
+			"aarch64-unknown-linux-musl":     true
+			"armv7-unknown-linux-gnueabihf":  true
+			"armv7-unknown-linux-musleabihf": true
+			"x86_64-apple-darwin":            true
+			"x86_64-pc-windows-msv":          true
+			"x86_64-unknown-linux-gnu":       true
+			"x86_64-unknown-linux-musl":      true
+		}
+		requirements: []
+		warnings: []
+		notices: []
+	}
+
+	installation: {
+		platform_name: null
+	}
+
+	configuration: {
```

Comment:
Since there are two different modes I am not sure how to do this elegantly. I could add a parallel enum (e.g. `ModeType`) similar to the current `Mode` just for specifying scheduled or streaming and then validate that it matches the `Mode` enum but that feels wrong. I could pull out the config from the enum and then match and validate on start up. I would love to keep the single combined enum and config but I can't think of a way to do that if the nested key can sometimes be scheduled and sometimes be streaming:-

`mode = scheduled
 scheduled.exec_interval_secs = 10`

or

`mode = scheduled
 streaming.respawn_on_exit = false`

I could do this but I am not sure that is what you want:-

`mode = scheduled
 mode_config.exec_interval_secs = 10`

or 

`mode = scheduled
 mode_config.respawn_on_exit = false`

Please let me know how you want to proceed with this (There might be a more elegant way but at the moment I am not seeing it).

Reply:
👍 My preference is for:

```
mode = scheduled
scheduled.exec_interval_secs = 10
```

Which is how the `aws_s3` source does it as well (given it only has one "mode" at the moment).

Reply:
Ok I will pull out the config from the Mode enum and have something like aws_s3 as follows:-

```
pub mode: Mode,
pub schedule: Option<ScheduledConfig>,
pub streaming: Option<StreamingConfig>,
```

I had been trying to avoid splitting out the config from the enum but I can't think of a way to do it.

Reply:
I created a issue request on the serde project related to this:- https://github.com/serde-rs/serde/issues/2013

Reply:
Changed to the new structure now.